wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/MediaInfoIdQueryAction.java

```diff
@@ -0,0 +1,96 @@
+package org.wikidata.wdtk.wikibaseapi;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import org.wikidata.wdtk.datamodel.helpers.Datamodel;
+import org.wikidata.wdtk.datamodel.interfaces.MediaInfoIdValue;
+import org.wikidata.wdtk.wikibaseapi.apierrors.MediaWikiApiErrorException;
+
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * Action for MediaInfoId retrieval.
+ *
+ * @author Lu Liu
+ */
+public class MediaInfoIdQueryAction {
+
+	private final ApiConnection connection;
+
+	private final String siteIri;
+
+	public MediaInfoIdQueryAction(ApiConnection connection, String siteIri) {
+		this.connection = connection;
+		this.siteIri = siteIri;
+	}
+
+	/**
+	 * Fetches the MediaInfoIds of files with the given names.
+	 * <p>
+	 * This method <b>only works with file name</b> (e.g. "File:Albert Einstein Head.jpg").
+	 * The "File:" prefix can be omitted, in this case, it will be automatically added during processing.
+	 * For example, "Albert Einstein Head.jpg" will be processed as "File:Albert Einstein Head.jpg".
+	 * <p>
+	 * Notice that pages other than file pages will also be fitted with the "File:" prefix.
+	 * For example, "Main Page" will be processed as "File:Main Page", which doesn't exist.
+	 * <b>So always make sure you are dealing with file name.</b>
+	 *
+	 * @param fileNames list of file names of the requested MediaInfoIds
+	 * @return map from file names for which data could be found to the MediaInfoIds
+	 * that were retrieved
+	 */
+	public Map<String, MediaInfoIdValue> getMediaInfoIds(List<String> fileNames)
+			throws IOException, MediaWikiApiErrorException {
+		// file name => file name with prefix
+		List<String> fileNamesWithPrefix = new ArrayList<>();
+		for (String fileName : fileNames) {
+			fileName = fileName.startsWith("File:") ? fileName : "File:" + fileName;
```

Comment:
`Image:` is also a possible namespace for files on Wikimedia Commons. I would also add support for it